src/builtins.c

```diff
@@ -1116,6 +1116,56 @@ JL_CALLABLE(jl_f_invoke)
     return res;
 }
 
+JL_CALLABLE(jl_f_invoke_kwsorter)
+{
+    JL_NARGSV(invoke, 3);
+    jl_value_t *kwargs = args[0];
+    // args[1] is `invoke` itself
+    jl_value_t *func = args[2];
+    jl_value_t *argtypes = args[3];
+    jl_value_t *kws = jl_get_keyword_sorter(func);
+    JL_GC_PUSH1(&argtypes);
+    if (jl_is_tuple(argtypes)) {
+        jl_depwarn("`invoke(f, (types...), ...)` is deprecated, "
```

Comment:
Unfortunately we do, since I transform the tuple type, so the following call to `jl_f_invoke` won't see that there was a tuple.